linker/shared/src/test/scala/org/scalajs/linker/OptimizerTest.scala

```diff
@@ -288,6 +289,81 @@ class OptimizerTest {
     testLink(classDefs, MainTestModuleInitializers)
   }
 
+  @Test
+  def testCaptureEliminationPreservesSideEffectsOrdering(): AsyncResult = await {
+    val sideEffect = m("sideEffect", List(I), I)
+
+    val x = LocalName("x")
+    val x2 = LocalName("x2")
+    val x4 = LocalName("x4")
+
+    val classDefs = Seq(
+        classDef(
+            MainTestClassName,
+            kind = ClassKind.Class,
+            superClass = Some(ObjectClass),
+            memberDefs = List(
+                trivialCtor(MainTestClassName),
+                // @noinline static def sideEffect(x: Int): Int = x
+                MethodDef(EMF.withNamespace(PublicStatic), sideEffect, NON,
+                    List(paramDef(x, IntType)), IntType, Some(VarRef(x)(IntType)))(
+                    EOH.withNoinline(true), None),
+                // static def main(args: String[]) {
+                //   console.log(arrow-lambda<
```

Comment:
```suggestion
                /* static def main(args: String[]) {
                 *   console.log(arrow-lambda<
```
etc.

We always use `/*` style comments for multiline comments.